packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -634,6 +646,7 @@ class _IOSDevicePortForwarder extends DevicePortForwarder {
       // TODO(ianh): This is a flakey race condition, https://github.com/libimobiledevice/libimobiledevice/issues/674
       connected = !await process.stdout.isEmpty.timeout(_kiProxyPortForwardTimeout, onTimeout: () => false);
       if (!connected) {
+        process.kill();
```

Comment:
Before this change, inside this while loop, we could spawn...enough `iproxy` instances here to exhaust the available file descriptors (as each tries to bind to a port).